bin/nrnivmodl_makefile_cmake.in

```diff
@@ -29,13 +29,39 @@ datadir_lib := ${ROOT}/@CMAKE_INSTALL_DATADIR@/lib
 # - @NRN_COMPILE_DEFS
 # - @NRN_LINK_DEFS
 LDFLAGS = $(LINKFLAGS) $(UserLDFLAGS) @NRN_LINK_DEFS@
-NRNLIB_FLAGS = -L$(libdir) -lnrniv
+
+# In GPU wheel distributions then the shipped libnrniv.so is linked against
+# some NVIDIA runtime libraries that are shipped with the wheel. If we use
+# nrnivmodl on the user machine then the NVIDIA compilers will link the local
+# versions of these libraries too, causing duplication.
+libnrniv_without_nvidia: $(libdir)/libnrniv.so
+	cp -v $(libdir)/libnrniv.so $(OUTPUT)/libnrniv-without-nvidia.so
+	patchelf $(OUTPUT)/libnrniv-without-nvidia.so --print-needed | grep '^libnv\(hpcatm\|omp\|cpumath\|cpumath-avx2\|c\)-[a-f0-9]\{8\}\.so' | xargs -t -r -n 1 patchelf $(OUTPUT)/libnrniv-without-nvidia.so --remove-needed
+	patchelf $(OUTPUT)/libnrniv-without-nvidia.so --set-soname libnrniv-without-nvidia.so
+	patchelf $(OUTPUT)/libnrniv-without-nvidia.so --print-rpath
+	ldd $(OUTPUT)/libnrniv-without-nvidia.so
```

Comment:
Is this last `ldd` extra?

Reply:
that doesn't sound appealing argument though if our goal is to reduce the clutter...

Reply:
This may not be the place/time to review the warnings and notices. But there is a problem that they obscure the substantive   messages. Many of the warnings/notices were added when the "feature" was added many years ago and changed previous behavior and were not so intrusive when only a few mod files were involved.

Reply:
I'm unsure what do here -- I agree that `nrnivmodl` is too noisy and that that should be improved, but I also think that, in the specific case that these messages get printed, that they are more useful than the the average line of output 😅. 

I would like to get this merged, so let me know what you want.

Reply:
I'm in favour of having `ldd` output for starters.

Reply:
In the current form, compiling `ringtest` gives:

```console
$ ./install/bin/nrnivmodl -coreneuron ../../ringtest/mod/
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/bbpv/repos/bbp/nrn/build_gpu
Mod files: "../../ringtest/mod/halfgap.mod"

Creating x86_64 directory for .o files.

COBJS=''
[INFO] Running: make -j4 -f /gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/bbpv/repos/bbp/nrn/build_gpu/install/share/coreneuron/nrnivmodl_core_makefile ROOT=/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/bbpv/repos/bbp/nrn/build_gpu/install MODS_PATH=x86_64/corenrn/mod2c BUILD_TYPE=STATIC NRN_PRCELLSTATE=0
Default NMODL flags:
Translating x86_64/corenrn/mod2c/exp2syn.mod into x86_64/corenrn/mod2c//exp2syn.cpp
Translating x86_64/corenrn/mod2c/halfgap.mod into x86_64/corenrn/mod2c//halfgap.cpp
Translating x86_64/corenrn/mod2c/expsyn.mod into x86_64/corenrn/mod2c//expsyn.cpp
cvode_emit=0 cvode_not_allowed=0
cvode_emit=0 cvode_not_allowed=0
cvode_emit=0 cvode_not_allowed=0
Thread Safe
Thread Safe
Thread Safe
diff: x86_64/corenrn/mod2c/_mod_func.cpp: No such file or directory
Translating x86_64/corenrn/mod2c/hh.mod into x86_64/corenrn/mod2c//hh.cpp
cvode_emit=0 cvode_not_allowed=0
Thread Safe
Translating x86_64/corenrn/mod2c/netstim.mod into x86_64/corenrn/mod2c//netstim.cpp
Translating x86_64/corenrn/mod2c/passive.mod into x86_64/corenrn/mod2c//passive.cpp
cvode_emit=0 cvode_not_allowed=0
Thread Safe
cvode_emit=0 cvode_not_allowed=0
Notice: ARTIFICIAL_CELL is a synonym for POINT_PROCESS which hints that it
only affects and is affected by discrete events. As such it is not
located in a section and is not associated with an integrator
Thread Safe
Translating x86_64/corenrn/mod2c/pattern.mod into x86_64/corenrn/mod2c//pattern.cpp
cvode_emit=0 cvode_not_allowed=0
Notice: ARTIFICIAL_CELL is a synonym for POINT_PROCESS which hints that it
only affects and is affected by discrete events. As such it is not
located in a section and is not associated with an integrator
Thread Safe
Translating x86_64/corenrn/mod2c/stim.mod into x86_64/corenrn/mod2c//stim.cpp
cvode_emit=0 cvode_not_allowed=0
Thread Safe
Translating x86_64/corenrn/mod2c/svclmp.mod into x86_64/corenrn/mod2c//svclmp.cpp
cvode_emit=0 cvode_not_allowed=0
Thread Safe
NVC++-W-1053-External and Static variables are not supported in acc routine - _ZN10coreneuron11nrn_threadsE (x86_64/corenrn/mod2c/exp2syn.cpp: 393)
NVC++/x86-64 Linux 22.2-0: compilation completed with warnings
NVC++-W-1053-External and Static variables are not supported in acc routine - _ZN10coreneuron11nrn_threadsE (x86_64/corenrn/mod2c/expsyn.cpp: 375)
NVC++/x86-64 Linux 22.2-0: compilation completed with warnings
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/bbpv/repos/bbp/nrn/build_gpu
 => Binary creating x86_64/special-core
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/bbpv/repos/bbp/nrn/build_gpu/install/share/coreneuron/coreneuron.cpp:
[INFO] MOD files built successfully for CoreNEURON
 -> Compiling mod_func.cpp
 -> NMODL ../../../ringtest/mod//halfgap.mod
'/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/bbpv/repos/bbp/nrn/build_gpu/install/lib/libnrniv.so' -> './libnrniv-without-nvidia.so'
Translating halfgap.mod into /gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/bbpv/repos/bbp/nrn/build_gpu/x86_64/halfgap.c
Thread Safe
 -> Compiling halfgap.c
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/bbpv/repos/bbp/nrn/build_gpu/install/lib:/gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_externals/install_gcc-11.2.0-skylake/python-3.9.7-yj5alh/lib:/gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_externals/install_gcc-11.2.0-skylake/hpe-mpi-2.25.hmpt-4ukyxt/lib:/gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_externals/install_gcc-11.2.0-skylake/nvhpc-22.2-gou6qx/Linux_x86_64/22.2/compilers/lib:/gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_compilers/install_gcc-4.8.5-haswell/gcc-11.2.0-suikmu/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../lib64
	linux-vdso.so.1 =>  (0x00007fffedb02000)
	libreadline.so.6 => /lib64/libreadline.so.6 (0x00007fffecef4000)
	libncurses.so.5 => /lib64/libncurses.so.5 (0x00007fffecccd000)
	libtinfo.so.5 => /lib64/libtinfo.so.5 (0x00007fffecaa3000)
	libform.so.5 => /lib64/libform.so.5 (0x00007fffec893000)
	libpython3.9.so.1.0 => /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_externals/install_gcc-11.2.0-skylake/python-3.9.7-yj5alh/lib/libpython3.9.so.1.0 (0x00007fffec4e2000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fffec2c6000)
	libmpi.so => /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_externals/install_gcc-11.2.0-skylake/hpe-mpi-2.25.hmpt-4ukyxt/lib/libmpi.so (0x00007fffebe95000)
	libatomic.so.1 => /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_compilers/install_gcc-4.8.5-haswell/gcc-11.2.0-suikmu/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../lib64/libatomic.so.1 (0x00007fffedad5000)
	libnvhpcatm.so => /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_externals/install_gcc-11.2.0-skylake/nvhpc-22.2-gou6qx/Linux_x86_64/22.2/compilers/lib/libnvhpcatm.so (0x00007fffebc8a000)
	libstdc++.so.6 => /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_compilers/install_gcc-4.8.5-haswell/gcc-11.2.0-suikmu/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../lib64/libstdc++.so.6 (0x00007fffeba76000)
	libnvomp.so => /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_externals/install_gcc-11.2.0-skylake/nvhpc-22.2-gou6qx/Linux_x86_64/22.2/compilers/lib/libnvomp.so (0x00007fffead74000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007fffeab70000)
	libnvcpumath.so => /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_externals/install_gcc-11.2.0-skylake/nvhpc-22.2-gou6qx/Linux_x86_64/22.2/compilers/lib/libnvcpumath.so (0x00007fffea72f000)
	libnvc.so => /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_externals/install_gcc-11.2.0-skylake/nvhpc-22.2-gou6qx/Linux_x86_64/22.2/compilers/lib/libnvc.so (0x00007fffea4ce000)
	libc.so.6 => /lib64/libc.so.6 (0x00007fffea100000)
	libgcc_s.so.1 => /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_compilers/install_gcc-4.8.5-haswell/gcc-11.2.0-suikmu/lib/gcc/x86_64-pc-linux-gnu/11.2.0/../../../../lib64/libgcc_s.so.1 (0x00007fffedab7000)
	libm.so.6 => /lib64/libm.so.6 (0x00007fffe9dfe000)
	libcrypt.so.1 => /lib64/libcrypt.so.1 (0x00007fffe9bc7000)
	libintl.so.8 => /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_externals/install_gcc-11.2.0-skylake/gettext-0.21-i2x2t4/lib/libintl.so.8 (0x00007fffedaa8000)
	libutil.so.1 => /lib64/libutil.so.1 (0x00007fffe99c4000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fffed8e3000)
	librt.so.1 => /lib64/librt.so.1 (0x00007fffe97bc000)
	libfreebl3.so => /lib64/libfreebl3.so (0x00007fffe95b9000)
	libiconv.so.2 => /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_externals/install_gcc-11.2.0-skylake/libiconv-1.16-hxxlex/lib/libiconv.so.2 (0x00007fffed9a7000)
 => LINKING shared library ./libnrnmech.so
 => LINKING executable ./special LDFLAGS are:     -lreadline -lcurses -lform /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_externals/install_gcc-11.2.0-skylake/python-3.9.7-yj5alh/lib/libpython3.9.so -Wl,-rpath,/gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_externals/install_gcc-11.2.0-skylake/python-3.9.7-yj5alh/lib -lpthread /gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_externals/install_gcc-11.2.0-skylake/hpe-mpi-2.25.hmpt-4ukyxt/lib/libmpi.so -Wl,-rpath,/gpfs/bbp.cscs.ch/ssd/apps/bsd/2022-01-10/stage_externals/install_gcc-11.2.0-skylake/hpe-mpi-2.25.hmpt-4ukyxt/lib
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/bbpv/repos/bbp/nrn/build_gpu/install/share/nrn/nrnmain.cpp:
Successfully created x86_64/special
```

I agree that this is helpful for `developer` when there `is an error` but if think as normal `neuron user` then I still feel it will confuse more for any neuron user as they are now seeing something new.

It will be useful for debugging when patchelf etc. fails and when user needs to provide some extra info to developers. In this case, we should think about some other strategy. (one example - have some extra env var like `NEURON_DEBUG=1 nrnivmodl ...` and that's going to dump enough info that will aid developers).

I will create a ticket for this.

Reply:
I'm rooting for `NEURON_DEBUG=1`. However I'm of two minds when it comes to the short term. I'd prefer to have tickets coming in directly with this information so as to avoid an extra `Hey, run with NEURON_DEBUG=1`. On the other hand, as you said, lots of stuff that we can chalk up to clutter (and I'd say it's debatable what information is `developer` friendly anyways :-)